src/binary-reader.cc

```diff
@@ -930,11 +930,16 @@ static void read_names_section(Context* ctx, uint32_t section_size) {
         uint32_t num_names;
         in_u32_leb128(ctx, &num_names, "name count");
         CALLBACK(OnFunctionNamesCount, num_names);
+        uint32_t last_function_index = 0;
         for (uint32_t j = 0; j < num_names; ++j) {
           uint32_t function_index;
           StringSlice function_name;
 
           in_u32_leb128(ctx, &function_index, "function index");
+          RAISE_ERROR_UNLESS(
```

Comment:
Hmm.. well technically the spec says "sorted by index" which a repeating sequence of zero indexes would be :)    This was enough to make sure my lld port was going the right thing.